lib/snapshot.go

```diff
@@ -394,10 +443,8 @@ func NewSnapshot(mainDb *badger.DB, mainDbDirectory string, snapshotBlockHeightP
 	//   Either way, it means our snapshot was compromised and we need to recompute it as described
 	//   in the previous bullet.
 	shouldRestart := false
-	if operationChannel.StateSemaphore > 0 || status.IsFlushing() {
+	if operationChannel.StateSemaphore > 0 {
```

Comment:
Sooo can StateSemaphore be > 0 now? I'm going to take a closer look at this after I work out just to see if we can eliminate this edge-case.